dot/state/block.go

```diff
@@ -322,6 +326,37 @@ func (bs *BlockState) SetBlockData(blockData *types.BlockData) error {
 	return err
 }
 
+// SetBlockDataAllFields call SetBlockDataSimple and then set all optional fields to the db
+func (bs *BlockState) SetBlockDataAllFields(blockData *types.BlockData) error {
```

Comment:
we should be setting all the fields separately

Reply:
we already have CompareAndSetBlockData: https://github.com/ChainSafe/gossamer/pull/709/files#diff-f3d5a6e9d6e7c92e678068b8316babd5R327

I think if we have that already we don't need this